### PR TITLE
Allow borderless `SearchBox.ExpandButton`

### PR DESCRIPTION
**Pull request by r100-stack**
## Changes

Goal of this PR is to allow a borderless `SearchBox.ExpandButton` during the collapsed state. This should be supported since `SearchBox.ExpandButton` is essentially an `IconButton`. However, it doesn't happen so far even with adding `styleType="borderless"` to `SearchBox.ExpandButton` since the border that on the container is never removed.

I first thought to add a new `styleType` prop to `CollapsedState`. When this prop is `borderless`, it sets a data attribute in `iui-expandable-searchbox`. Presence of this attribute will remove its border and the background. However, to decrease the complexity of the API, @mayank99 suggested to try to avoid adding a new prop.

The current approach I'm using is to directly hide the border and background on `iui-expandable-searchbox` when it's collapsed. No visual change should be perceived as the same background and border now come from the `ExpandButton` instead.

<details>
<summary>Collapsed state (before and after are the same)</summary>

screenshot

</details>

Users using the `SearchBox` in ways that we haven't demoed in our examples may notice some change after this PR. However, after discussing with @mayank99, we concluded that those custom cases are uncommon and also unlikely to be useful for users.

## Testing

Added a story and thus a visual test. Since this PR mainly adds a visual change using css styles, I felt no unit or e2e tests were necessary.

## Docs

* Added one minor changeset each for `@itwin/itwinui-css` and `@itwin/itwinui-react`.
* Added docs and an accompanying demo.

### CODE REVIEW

File: examples/SearchBox.borderlessexpandbutton.css [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.demo-container {
+  width: 70%;
+  display: flex;
+  justify-content: center;
+  flex-direction: column;
```
[the remark this answers is not in the document]
Comment (mayank99) [reply examples/SearchBox.borderlessexpandbutton.css:5]: center looks good to me

File: packages/itwinui-css/src/searchbox/searchbox.scss [hunk 2]
```diff
@@ -5,11 +5,21 @@
...
   }
+
+  &:not([data-iui-expanded='true']) {
+    &:is(:hover, :not(:hover)) {
+      --_iui-input-background-color: transparent;
+    }
+
+    &::before {
+      transition: none;
+      --_iui-input-border-color: transparent;
```
Comment (mayank99) [inline packages/itwinui-css/src/searchbox/searchbox.scss:20]: Do we still need the transparent border?

File: packages/itwinui-css/src/searchbox/searchbox.scss [hunk 3]
```diff
@@ -5,11 +5,21 @@
...
   padding-inline: 0; // To remove 1px padding from input-flex-container
 
-  // Expanded
   &:where([data-iui-expanded='true']) {
     inline-size: 100%;
     border-color: var(--_iui-input-border-color);
   }
+
+  &:not([data-iui-expanded='true']) {
+    &:is(:hover, :not(:hover)) {
```
Comment (mayank99) [inline packages/itwinui-css/src/searchbox/searchbox.scss:14]: This selector shouldn't be necessary I think? Are we not able to just use `background-color: transparent` directly?

File: examples/SearchBox.borderlessexpandbutton.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (smmr-dn) [reply examples/SearchBox.borderlessexpandbutton.css]: I will also go with center.

**Review by mayank99 — APPROVED** [review]
LGTM

**Review by smmr-dn — APPROVED** [review]

**Pull request merged**